SharpSource/SharpSource/Diagnostics/AsyncOverloadsAvailableAnalyzer.cs

```diff
@@ -27,84 +25,69 @@ public override void Initialize(AnalysisContext context)
     {
         context.EnableConcurrentExecution();
         context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
-        context.RegisterSyntaxNodeAction(AnalyzeSyntaxNode, SyntaxKind.InvocationExpression);
-    }
-
-    private void AnalyzeSyntaxNode(SyntaxNodeAnalysisContext context)
-    {
-        var invocation = (InvocationExpressionSyntax)context.Node;
-        var surroundingDeclaration = invocation.FirstAncestorOrSelfOfType(SyntaxKind.MethodDeclaration, SyntaxKind.GlobalStatement, SyntaxKind.SimpleLambdaExpression);
-
-        var isInCorrectContext = surroundingDeclaration switch
-        {
-            MethodDeclarationSyntax method => method.Modifiers.Any(SyntaxKind.AsyncKeyword),
-            GlobalStatementSyntax => true,
-            SimpleLambdaExpressionSyntax lambda => lambda.Modifiers.Any(SyntaxKind.AsyncKeyword),
-            _ => false
-        };
 
-        if (!isInCorrectContext || surroundingDeclaration == default)
+        context.RegisterCompilationStartAction(compilationContext =>
         {
-            return;
-        }
-
-        switch (invocation.Expression)
-        {
-            case MemberAccessExpressionSyntax memberAccess:
-                CheckIfOverloadAvailable(memberAccess.Name, context, surroundingDeclaration, invocation);
-                break;
-            case IdentifierNameSyntax identifierName:
-                CheckIfOverloadAvailable(identifierName, context, surroundingDeclaration, invocation);
-                break;
-            case GenericNameSyntax genericName:
-                CheckIfOverloadAvailable(genericName, context, surroundingDeclaration, invocation);
-                break;
-            default:
-                break;
-        }
+            var cancellationTokenSymbol = compilationContext.Compilation.GetTypeByMetadataName("System.Threading.CancellationToken");
+            if (cancellationTokenSymbol is not null)
```

Comment:
Does this symbol have to exist for the analyzer to report a diagnostic? If not, I think you can just proceed even with a null symbol